2_shoppingMall_SPA/src/pages/ProductListPage.js

```diff
@@ -0,0 +1,58 @@
+import { getProductsList } from '../api/api.js';
+import LocalStorageUtil from '../utils/localStorage.js';
+import { routeChange } from '../utils/route.js';
+
+export default function ProductListPage({ $target }) {
+    // product = {
+    //     id: number;
+    //     name: string;
+    //     imageUrl: string;
+    //     price: number; 
+    // }
+    this.$page;
+    this.productList = [];
+
+    this.init = async () => {
+        this.$page = document.createElement('div')
+        this.$page.className = 'ProductListPage'
+        $target.appendChild(this.$page);
+      
+        this.productList = LocalStorageUtil.getItem('productList');
+        
+        if(this.productList == null) {
+            this.productList = await getProductsList({
+                setLoading: () => {}, 
+                finishLoading: () => {},
+            });
```

Comment:
```suggestion
            this.productList = await getProductsList();
```
`getProductsList` 함수는 인자를 받지 않는 것으로 설계되어 있습니다. 별도의 인자를 전달하지 않고 호출해야 합니다. 🥲